includes/kaltura_derivatives.inc

```diff
@@ -5,12 +5,36 @@
  * @param AbstractObject $object An Islandora Object
  */
 function islandora_video_streaming_kaltura_upload($object, $force = false) {
-  if ($force || !isset($object['KALTURA_STREAM'])) {
+    //If the object's KALTURA_STREAM is null then we add it into the file system
+    //using islandora_video_copy_obj_datastream
+    if ($force || !isset($object['KALTURA_STREAM'])) {
     $archival_file = islandora_video_copy_obj_datastream($object);
     if ($archival_file === FALSE) {
       return islandora_video_no_obj_datastream($object->id);
     }
-    $filePath = drupal_realpath($archival_file['file']->uri);
+    
+    $old_content= null; 
+    $old_entity_id = null;
+    //if an existing object['KALTURA_STREAM'] is present, delete existing kaltura
+    //video and media entry. 
+    if (isset($object['KALTURA_STREAM']))
+    {
+        //fetch actual datastream content FROM islandora
+        $old_content = $object['KALTURA_STREAM']->content;
+        //if $old_content exists then delete it from Kaltura server
+        if (isset($old_content))
+        {
+            //decode $old_content for entry_id
+            $decoded = json_decode($old_content, true);
+            //find entity through $decoded string 
+            $entity = kaltura_entry_load($decoded['entry_id']);
```

Comment:
What if the KALTURA_STREAM exists, but is a bad or broken format (invalid JSON)?  You don't have any error checking on fetching `$decoded`, `$entity`, or `$old_entity_id`.